addons/escoria-core/game/core-scripts/esc/esc_object_manager.gd

```diff
@@ -207,6 +210,10 @@ func register_object(object: ESCObject, room: ESCRoom = null, force: bool = fals
 
 	var objects: Dictionary = _get_room_objects_objects(room_key)
 	objects[object.global_id] = object
+	
+	# If object state is not STATE_DEFAULT, save it in manager's object states 
+	if object.state != ESCObject.STATE_DEFAULT:
+		get_object(object.global_id).state = object.state
```

Comment:
There needs to be a null check based on what `get_object` returns, since returning null is a possibility.